### PR TITLE
fix: updated dependencies and resolved all upcoming issues

### PR DESCRIPTION
**Pull request by rrieger-espirit**
updated dependencies + resolving issues

### CODE REVIEW

File: src/components/BaseComponent.tsx [hunk 1]
```diff
@@ -5,9 +5,9 @@ const breakpoints = require("tailwindcss/defaultTheme");
 
 @Component
 class BaseComponent<
-  Props = {},
-  EventsWithOn = {},
-  Slots = {}
+  Props = unknown,
```
Comment (henczi-espirit) [inline src/components/BaseComponent.tsx:8]: What is "unknown" in this context?

File: src/components/ImageSlider/index.tsx [hunk 2]
```diff
@@ -7,7 +7,7 @@ import Slider from "../Slider";
 @Component({ name: "ImageSlider" })
 class ImageSlider<Type = any> extends BaseComponent<
   ImageSliderProps<Type>,
-  {},
+  unknown,
```
Comment (henczi-espirit) [inline src/components/ImageSlider/index.tsx:10]: what is "unknown" in this context?

File: src/components/NewsDetail/index.tsx [hunk 3]
```diff
@@ -13,7 +13,11 @@ import { Component, Prop } from "vue-property-decorator";
 @Component({
   name: "NewsDetail",
 })
-class NewsDetail extends BaseComponent<NewsDetailProps, {}, NewsDetailSlots> {
+class NewsDetail extends BaseComponent<
+  NewsDetailProps,
+  unknown,
```
Comment (henczi-espirit) [inline src/components/NewsDetail/index.tsx:18]: what is "unknown" in this context?

File: src/components/Sections/GoogleMapsSection/index.tsx [hunk 4]
```diff
@@ -93,7 +93,7 @@ const DEFAULT_STYLES: google.maps.MapTypeStyle[] = [
 })
 class GoogleMapsSection extends BaseComponent<
   GoogleMapsSectionProps,
-  {},
+  unknown,
```
Comment (henczi-espirit) [inline src/components/Sections/GoogleMapsSection/index.tsx:96]: what is "unknown" in this context?

File: src/components/Sections/InterestingFactsSection/index.tsx [hunk 5]
```diff
@@ -17,7 +17,7 @@ import {
 })
 class InterestingFactsSection extends BaseComponent<
   InterestingFactsSectionProps,
-  {},
+  unknown,
```
Comment (henczi-espirit) [inline src/components/Sections/InterestingFactsSection/index.tsx:20]: what is "unknown" in this context?

File: src/components/Sections/ListSection/index.tsx [hunk 6]
```diff
@@ -10,7 +10,7 @@ import Button from "@/components/Button";
 })
 class ListSection<Item = any> extends BaseComponent<
   ListSectionProps<Item>,
-  {},
+  unknown,
```
Comment (henczi-espirit) [inline src/components/Sections/ListSection/index.tsx:13]: what is "unknown" in this context?

File: src/components/Sections/NewsTeaserSection/index.tsx [hunk 7]
```diff
@@ -14,7 +14,7 @@ const ImageSrc = require("./../../../assets/dots.png");
 })
 class NewsTeaserSection extends BaseComponent<
   NewsTeaserSectionProps,
-  {},
+  unknown,
```
Comment (henczi-espirit) [inline src/components/Sections/NewsTeaserSection/index.tsx:17]: what is "unknown" in this context?

File: src/components/Sections/ListSection/index.tsx [hunk 8]
```diff
@@ -77,7 +77,9 @@ class ListSection<Item = any> extends BaseComponent<
           this.$slots.default
         ) : (
           <div class="ui-grid ui-grid-cols-1 ui-gap-4 sm:ui-grid-cols-2 md:ui-grid-cols-3 lg:ui-grid-cols-4 xl:ui-gap-8">
-            {this.items.map(item => this.$scopedSlots.item(item))}
+            {this.items.map((item) => {
+              return this.$scopedSlots.item(item as Exclude<Item, undefined>);
```
Comment (henczi-espirit) [inline src/components/Sections/ListSection/index.tsx:81]: Why do we need this "return" statement?

File: src/components/Sections/ProductDetailSection/index.tsx [hunk 9]
```diff
@@ -17,7 +17,7 @@ import ImageSlider from "@/components/ImageSlider";
 })
 class ProductDetailSection extends BaseComponent<
   ProductDetailSectionProps,
-  {},
+  unknown,
```
Comment (henczi-espirit) [inline src/components/Sections/ProductDetailSection/index.tsx:20]: what is "unknown" in this context?

File: src/components/Slider/__tests__/Slider.spec.tsx [hunk 10]
```diff
@@ -5,7 +5,7 @@ import { Component } from "vue-property-decorator";
 
 describe("components/slider", () => {
   @Component
-  class TestSlide extends BaseComponent<{}> {
+  class TestSlide extends BaseComponent<unknown> {
```
Comment (henczi-espirit) [inline src/components/Slider/__tests__/Slider.spec.tsx:8]: what is "unknown" in this context?

File: src/components/Slider/index.tsx [hunk 11]
```diff
@@ -9,7 +9,7 @@ const defaultAnimation = () => {
 @Component({
   name: "Slider",
 })
-class Slider extends BaseComponent<SliderProps, {}, SliderSlots> {
+class Slider extends BaseComponent<SliderProps, unknown, SliderSlots> {
```
Comment (henczi-espirit) [inline src/components/Slider/index.tsx:12]: what is "unknown" in this context?

File: src/components/internal/Documentation/ComponentWrapper/index.tsx [hunk 12]
```diff
@@ -12,7 +12,11 @@ interface ComponentWrapperSlots {
 @Component({
   name: "ComponentWrapper",
 })
-class ComponentWrapper extends BaseComponent<{}, {}, ComponentWrapperSlots> {
+class ComponentWrapper extends BaseComponent<
+  unknown,
```
Comment (henczi-espirit) [inline src/components/internal/Documentation/ComponentWrapper/index.tsx:16]: what is "unknown" in this context?

File: src/components/utils/PropertyChecker/__tests__/PropertyChecker.spec.ts [hunk 13]
```diff
@@ -13,9 +13,9 @@ describe("utils/PropertyChecker/checkPassedValue", () => {
     testingArray = ["a", "b", "c"];
     testingValue = "d";
     mockedFunction = jest.fn();
-    mockedVueElement = ({
+    mockedVueElement = {
       remove: mockedFunction,
-    } as unknown) as Element;
+    } as unknown as Element;
```
Comment (henczi-espirit) [inline src/components/utils/PropertyChecker/__tests__/PropertyChecker.spec.ts:18]: Why do we need 2x "as" here?

File: src/documentation/examples/Sections/FullWidthSliderSection.Images.slots.tsx [hunk 14]
```diff
@@ -11,42 +11,44 @@ class App extends Vue {
...
+            button: (props) => (
               <button
                 class="Button Button--variant--inverted ui-rounded"
                 onClick={props.onClick}
               >
                 {props.content}
               </button>
             ),
-            media: slideMedia => (
+            media: (slideMedia: any) => (
```
Comment (henczi-espirit) [inline src/documentation/examples/Sections/FullWidthSliderSection.Images.slots.tsx:30]: Why do we need the "any" type for the slideMedia, but not for the other attributes?

File: src/documentation/examples/Sections/ListSection.CustomContent.tsx [hunk 15]
```diff
@@ -34,12 +34,12 @@ export default class App extends Vue {
...
           items={[{ headline: "test" }]}
           filters={filters}
-          handleFilterChange={selectedFilters =>
+          handleFilterChange={(selectedFilters) =>
             (this.selectedFilters = selectedFilters)
           }
           selectedFilters={this.selectedFilters}
           scopedSlots={{
-            item: props => <div>{props.headline}</div>,
+            item: (props: any) => <div>{props.headline}</div>,
```
Comment (henczi-espirit) [inline src/documentation/examples/Sections/ListSection.CustomContent.tsx:42]: Why do we need the "any" type here?

File: src/documentation/examples/Sections/ListSection.CustomItem.tsx [hunk 16]
```diff
@@ -79,18 +75,18 @@ export default class App extends Vue {
         items={items}
         filters={filters}
         selectedFilters={this.selectedFilters}
-        handleFilterChange={selectedFilters =>
+        handleFilterChange={(selectedFilters) =>
           (this.selectedFilters = selectedFilters)
         }
         scopedSlots={{
-          item: item => (
+          item: (item: any) => (
```
Comment (henczi-espirit) [inline src/documentation/examples/Sections/ListSection.CustomItem.tsx:82]: Why do we need the "any" type here?

File: src/documentation/examples/Sections/ListSection.NoHeadlineAndFilters.tsx [hunk 17]
```diff
@@ -61,7 +57,7 @@ export default class App extends Vue {
       <ListSection
         items={items}
         scopedSlots={{
-          item: item => (
+          item: (item: any) => (
```
Comment (henczi-espirit) [inline src/documentation/examples/Sections/ListSection.NoHeadlineAndFilters.tsx:60]: Why do we need the "any" type here?

File: src/documentation/examples/Sections/ListSection.tsx [hunk 18]
```diff
@@ -96,11 +92,11 @@ export default class App extends Vue {
         items={items}
         filters={filters}
         selectedFilters={this.selectedFilters}
-        handleFilterChange={selectedFilters =>
+        handleFilterChange={(selectedFilters) =>
           (this.selectedFilters = selectedFilters)
         }
         scopedSlots={{
-          item: item => (
+          item: (item: any) => (
```
Comment (henczi-espirit) [inline src/documentation/examples/Sections/ListSection.tsx:99]: Why do we need the "any" type here?

File: src/types/components.d.ts [hunk 19]
```diff
@@ -476,12 +476,12 @@ export interface NewsDetailSlots {
...
    * It will receive the social-area-data as parameter
    */
-  social: SocialArea;
+  social?: SocialArea;
 }
 
 export class NewsDetail extends Component<
   NewsDetailProps,
-  {},
+  unknown,
```
Comment (henczi-espirit) [inline src/types/components.d.ts:484]: what is "unknown" in this context?

File: src/types/components.d.ts [hunk 20]
```diff
@@ -621,7 +621,7 @@ export interface SliderSlots {
     params: SliderControlParams;
   };
 }
-export class Slider extends Component<SliderProps, {}, SliderSlots> {}
+export class Slider extends Component<SliderProps, unknown, SliderSlots> {}
```
Comment (henczi-espirit) [inline src/types/components.d.ts:624]: what is "unknown" in this context?

File: src/types/components.d.ts [hunk 21]
```diff
@@ -651,6 +651,6 @@ export interface ImageSliderSlots<ImageType> {
 }
 export class ImageSlider<ImageType = ImageRef> extends Component<
   ImageSliderProps<ImageType>,
-  {},
+  unknown,
```
Comment (henczi-espirit) [inline src/types/components.d.ts:654]: what is "unknown" in this context?

File: src/types/internal.d.ts [hunk 22]
```diff
@@ -28,4 +28,4 @@ export interface DevInfoProps {
 }
 export class DevInfo extends Component<DevInfoProps> {}
 
-export class DevInfoTarget extends Component<{}> {}
+export class DevInfoTarget extends Component<unknown> {}
```
Comment (henczi-espirit) [inline src/types/internal.d.ts:31]: what is "unknown" in this context?

File: src/types/sections.d.ts [hunk 23]
```diff
@@ -121,7 +121,7 @@ export interface ProductDetailSectionSlots {
 
 export class ProductDetailSection extends Component<
   ProductDetailSectionProps,
-  {},
+  unknown,
```
Comment (henczi-espirit) [inline src/types/sections.d.ts:124]: what is "unknown" in this context?

File: src/types/sections.d.ts [hunk 24]
```diff
@@ -182,11 +182,11 @@ export interface InterestingFactsSectionProps {
 }
 export class InterestingFactsSection extends Component<
   InterestingFactsSectionProps,
-  {},
+  unknown,
```
Comment (henczi-espirit) [inline src/types/sections.d.ts:185]: what is "unknown" in this context?

File: tsconfig.json [hunk 25]
```diff
@@ -21,6 +22,7 @@
     "lib": ["esnext", "dom", "dom.iterable", "scripthost"]
   },
   "include": [
+    "node_modules/vue-tsx-support/enable-check.d.ts",
```
Comment (henczi-espirit) [inline tsconfig.json:25]: Why do we need this?

File: .eslintrc.js [hunk 26]
```diff
@@ -20,6 +20,8 @@ module.exports = {
     "@typescript-eslint/camelcase": "off",
     "@typescript-eslint/no-explicit-any": "off",
     "@typescript-eslint/no-non-null-assertion": "off",
+    "@typescript-eslint/explicit-function-return-type": "off",
```
Comment (henczi-espirit) [inline .eslintrc.js:23]: Isn't it better to have this enabled?

File: .eslintrc.js [hunk 27]
```diff
@@ -20,6 +20,8 @@ module.exports = {
...
+    "@typescript-eslint/explicit-module-boundary-types": "off",
```
Comment (henczi-espirit) [inline .eslintrc.js:24]: Isn't it better to have this enabled?

**Comment by dbarczewski** [comment]
PR will be closed since the dependencies were updated in #83

**Pull request closed without merging**